edge-functions/logging/pages/_middleware.ts

```diff
@@ -0,0 +1,27 @@
+import { NextRequest } from 'next/server'
+
+// Regex for public files
+const PUBLIC_FILE = /\.(.*)$/
+
+export default function middleware(req: NextRequest) {
+  // Only log for visited pages
+  if (!PUBLIC_FILE.test(req.nextUrl.pathname)) {
+    // We just fire and forget this request as we don't want to block the request until completion
```

Comment:
```suggestion
    // We fire and forget this request to avoid blocking the request until completion
    // and let logging occur in the background
```